File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n = height.size();
        int l=0,r=n-1,maxi=(n-1)*(min(height[0],height[n-1])),w=n-1;
        while(l<r){
            if (height[l]<height[r]) l++;
            else r--;
            w--;
            maxi=max(maxi,w*min(height[l],height[r]));
        }
        return maxi;
    }
};

File: Data Structures & Algorithms/balanced-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:

    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        if not root:
            return True

        def height(node):
            if not node:
                return 0
            return 1+max(height(node.left),height(node.right))

        q = deque([root])
        while q:
            node = q.pop()
            if height(node.left) - height(node.right) > 1 or height(node.left) - height(node.right) <-1:
                return False
            if node.left:
                q.append(node.left)
            if node.right:
                q.append(node.right)

        return True
        